tests/FileExtendedStatus.hs

```diff
@@ -0,0 +1,159 @@
+{-# OPTIONS_GHC -fno-warn-missing-signatures #-}
+
+module FileExtendedStatus (main) where
```

Comment:
Mostly a bold copy-pasta of `FileStatus.hs` with some adjustments, but I'm not really bullish of deduplication. Tests are stupid code.